components/gauge/include/bq27441_defs.h

```diff
@@ -0,0 +1,117 @@
+#ifndef _BQ27441_DEFS_H_
+#define _BQ27441_DEFS_H_
+
+/**
+ * @file bq27441_defs.h
+ * @brief Definitions for the BQ27441 fuel gauge.
+ */
+
+#define BQ27441_DATA_SIZE 3 * sizeof(uint16_t) /**< Size of the buffer to hold capacity, SoC, and voltage data*/
+
+/**
+ * @brief Sensor interface functions for BQ27441.
+ */
+extern sensor_interface_t bq27441_interface;
```

Comment:
The concept of interfaces is ok, but with externs not really... I'd suggest using some set/get api:
```
bme [source]
static sensor_interface_t bme280_interface = {some init}
sensor_interface_t *bme280_get_interface(void){return &bme280_interface;}
bq [source]
static sensor_interface_t bq27441_interface = {some init}
sensor_interface_t *bq24771_get_interface(void){return &bq27441_interface;}
```
Then in main you can call this in init for sensors.
`node.sensors.bq27441.bq_sensor_context.interface = bq24771_get_interface();`